pv_site_api/enode_auth.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+import httpx
+
+
+def get_enode_access_token() -> str:
+    # TODO: Add environment variables
+    client_id = os.getenv("CLIENT_ID")
+    client_secret = os.getenv("CLIENT_SECRET")
+    url = os.getenv("ENODE_TOKEN_URL")
+
+    # Encode the client_id and client_secret using Basic Auth
+    auth = httpx.BasicAuth(username=client_id, password=client_secret)
+
+    data = {"grant_type": "client_credentials"}
+
+    response = httpx.post(url, auth=auth, data=data)
+
+    return response.json()["access_token"]
+
+
+class EnodeAuth(httpx.Auth):
+    def __init__(self):
+        self.access_token = get_enode_access_token()
+
+    def auth_flow(self, request):
+        # Add the Authorization header to the request using the current access token
+        request.headers["Authorization"] = f"Bearer {self.access_token}"
+        response = yield request
+
+        if response.status_code == 401:
+            # The access token is no longer valid, refresh it
+            self.access_token = get_enode_access_token()
```

Comment:
I'm not entirely sure if this will work in the way HTTPX wants it to... consider the example they have in the docs: https://www.python-httpx.org/advanced/#customizing-authentication:~:text=refresh_response%20%3D%20yield%20self.build_refresh_request()

The way they do this is have the function they use to fetch the new token _return_ a Request object, which they then call `yield` on. I think we'll want to do the same thing here. This also means that you'll need to grab the `.json()` and the right key from the response in this code rather than your other function.